src/net/dns.rs

```diff
@@ -1,4 +1,44 @@
 //! DNS resolution and cache.
+//!
+//! DNS cache in Delta Chat has two layers:
+//! in-memory cache and persistent `dns_cache` SQL table.
+//!
+//! In-memory cache is using a "stale-while-revalidate" strategy.
+//! If there is a cached value, it is returned immediately
+//! and revalidation task is started in the background
+//! to replace old cached IP addresses with new ones.
+//! If there is no cached value yet,
+//! lookup only finishes when `lookup_host` returns first results.
+//! In-memory cache is shared between all accounts
+//! and is never stored on the disk.
+//! It can be thought of as an extension
+//! of the system resolver.
+//!
+//! Persistent `dns_cache` SQL table is used to collect
+//! all IP addresses ever seen for the hostname
+//! together with the timestamp
+//! of the last time IP address has been seen.
+//! Note that this timestamp reflects the time
+//! IP address was returned by the in-memory cache
+//! rather than the underlying system resolver.
+//! Unused entries are removed after 30 days
+//! (`CACHE_TTL` constant) to avoid having
+//! old non-working IP addresses in the cache indefinitely.
```

Comment:
IIUC, this PR pessimizes the following case: 
- all the IP addresses of the server aren't working anymore
- the server publishes a new IP address in the DNS
- a client that's already running tries to send a message
- -> with this PR, the client won't manage to connect, since only IP addresses from persistent and in-memory cache are tried out
- -> with this PR, the new IP address isn't immediately written into the persistent cache (next time the client tries to connect, it will)

I just wanted to point this out, you can probably tell better than me whether this is a problem.